src/index.tsx

```diff
@@ -129,51 +136,85 @@ class ApiConnection {
   async _start() {
     while (true) {
       await sleepMsec(17000);
-      this.sendMessage({type: 'keepAlive'});
+      if (!this._isDisconnected) this.sendMessage({type: 'keepAlive'});
     }
   }
 }
-const apiConnection = new ApiConnection();
-const baseSha1Url = `http://${window.location.hostname}:15309/sha1`;
-const hither = initializeHitherInterface((msg) => apiConnection.sendMessage(msg), baseSha1Url)
 
-apiConnection.onConnect(() => {
-  console.info('Connected to API server');
-})
-apiConnection.onMessage(msg => {
-  const type0 = msg.type;
-  if (type0 === 'reportServerInfo') {
-    theStore.dispatch({
-      type: SET_SERVER_INFO,
-      serverInfo: msg.serverInfo
-    });
-  }
-  else if (type0 === 'action') {
-    let action = msg.action;
-    if ('persistKey' in action) {
-      // just to be safe
-      delete action['persistKey'];
+const createApiConnection = () => {
```

Comment:
This code is a reorganized version of prior code, enabling reconnecting to the api server (websocket).